dynamic_rest/viewsets.py

```diff
@@ -418,36 +441,150 @@ def _bulk_update(self, data, partial=False):
+        )
+        return Response({
+            'meta': {
+                'updated': updated
+            }
+        }, status=status.HTTP_200_OK)
+
     def update(self, request, *args, **kwargs):
-        """Either update  a single or many model instances. Use list to
-        indicate bulk update.
+        """Update one or more model instances.
+
+        If ENABLE_BULK_UPDATE is set, multiple previously-fetched records
+        may be updated in a single call, provided their IDs.
+
+        If ENABLE_PATCH_ALL is set, multiple records
+        may be updated in a single PATCH call, even without knowing their IDs.
+
+        *WARNING*: ENABLE_PATCH_ALL should be considered an advanced feature
+        and used with caution. This feature must be enabled at the viewset level
+        and must also be requested explicitly by the client
+        via the "patch-all" query parameter.
+
+        This parameter can have one of the following values:
+
+            true (or 1): records will be fetched and then updated in a transaction loop
+              - The `Model.save` method will be called and model signals will run
+              - This can be slow if there are too many signals or many records in the query
+              - This is considered the more safe and default behavior
+            query: records will be updated in a single query
+              - The `QuerySet.update` method will be called and model signals will not run
+              - This will be fast, but may break data constraints that are controlled by signals
+              - This is considered unsafe but useful in certain situations
+
+        The server's successful response to a patch-all request
+        will NOT include any individual records. Instead, the response content will contain
+        a "meta" object with an "updated" count of updated records.
 
         Examples:
 
-        PATCH /dogs/1/
-        {
-            'fur': 'white'
-        }
+        Update one dog:
 
-        PATCH /dogs/
-        {
-            'dogs': [
+            PATCH /dogs/1/
+            {
+                'fur': 'white'
+            }
+
+        Update many dogs by ID:
+
+            PATCH /dogs/
+            [
                 {'id': 1, 'fur': 'white'},
                 {'id': 2, 'fur': 'black'},
                 {'id': 3, 'fur': 'yellow'}
             ]
-        }
 
-        PATCH /dogs/?filter{fur.contains}=brown
-        [
-            {'id': 3, 'fur': 'gold'}
-        ]
-        """
+        Update all dogs in a query:
+
+            PATCH /dogs/?filter{fur.contains}=brown&patch-all=true
+            {
+                'fur': 'gold'
+            }
+        """  # noqa
         if self.ENABLE_BULK_UPDATE:
-            partial = 'partial' in kwargs
-            bulk_payload = self._get_bulk_payload(request)
-            if bulk_payload:
-                return self._bulk_update(bulk_payload, partial)
+            patch_all = self.get_request_patch_all()
+            if (
```

Comment:
style: line wrapping probably not necessary here

Reply:
:+1:, I think this initially had longer names and one extra clause so it was too long to fit on one line